Core/GDCore/IDE/Events/ProjectElementRenamer.cpp

```diff
@@ -0,0 +1,219 @@
+      : platform(platform_), globalObjectsContainer(globalObjectsContainer_),
+        objectsContainer(objectsContainer_),
+        expressionPlainString(expressionPlainString_),
+        parameterType(parameterType_), objectName(objectName_),
+        layerName(layerName_), oldName(oldName_){};
+  virtual ~ExpressionIdentifierStringFinder(){};
+
+  const std::vector<gd::ExpressionParserLocation> GetOccurrences() const {
+    return occurrences;
+  }
+
+protected:
+  void OnVisitSubExpressionNode(SubExpressionNode &node) override {
+    node.expression->Visit(*this);
+  }
+  void OnVisitOperatorNode(OperatorNode &node) override {
+    node.leftHandSide->Visit(*this);
+    node.rightHandSide->Visit(*this);
+  }
+  void OnVisitUnaryOperatorNode(UnaryOperatorNode &node) override {
+    node.factor->Visit(*this);
+  }
+  void OnVisitNumberNode(NumberNode &node) override {}
+  void OnVisitTextNode(TextNode &node) override {}
+  void OnVisitVariableNode(VariableNode &node) override {
+    if (node.child)
+      node.child->Visit(*this);
+  }
+  void OnVisitVariableAccessorNode(VariableAccessorNode &node) override {
+    if (node.child)
+      node.child->Visit(*this);
+  }
+  void OnVisitVariableBracketAccessorNode(
+      VariableBracketAccessorNode &node) override {
+    node.expression->Visit(*this);
+    if (node.child)
+      node.child->Visit(*this);
+  }
+  void OnVisitIdentifierNode(IdentifierNode &node) override {}
+  void OnVisitObjectFunctionNameNode(ObjectFunctionNameNode &node) override {}
+  void OnVisitFunctionCallNode(FunctionCallNode &node) override {
+    bool considerFunction = objectName.empty() || node.objectName == objectName;
+
+    const bool isObjectFunction = !node.objectName.empty();
+    const gd::ExpressionMetadata &metadata =
+        isObjectFunction
+            ? MetadataProvider::GetObjectAnyExpressionMetadata(
+                  platform,
+                  GetTypeOfObject(globalObjectsContainer, objectsContainer,
+                                  node.objectName),
+                  node.functionName)
+            : MetadataProvider::GetAnyExpressionMetadata(platform,
+                                                         node.functionName);
+
+    if (gd::MetadataProvider::IsBadExpressionMetadata(metadata)) {
+      return;
+    }
+
+    gd::String lastLayerName;
+    size_t parameterIndex = 0;
+    for (size_t metadataIndex = (isObjectFunction ? 1 : 0);
+         metadataIndex < metadata.parameters.size() &&
+         parameterIndex < node.parameters.size();
+         ++metadataIndex) {
+      auto &parameterMetadata = metadata.parameters[metadataIndex];
+      if (parameterMetadata.IsCodeOnly()) {
+        continue;
+      }
+      auto &parameterNode = node.parameters[parameterIndex];
+      ++parameterIndex;
+
+      if (parameterMetadata.GetType() == "layer") {
+        // Remove quotes, it won't match if it's not a literal anyway.
+        lastLayerName = expressionPlainString.substr(
```

Comment:
It's like a look ahead. If the tree was like a DOM, it would have checked if the child was a literal string and read its content, but I'm not sure we can do this.